src/JobsPage.php

```diff
@@ -0,0 +1,139 @@
+    public const SLUG = 'scripted_jobs';
+
+    /**
+     * Builds an ajax preview url for the given job.
+     *
+     * @param  object $job
+     * @return string
+     */
+    public static function getJobAjaxPreviewUrl($job)
+    {
+        return sprintf(
+            '%s&action=%s&projectId=%s',
+            wp_nonce_url(admin_url('admin-ajax.php'), JobTasks::AJAX_FINISHED_JOB_PREVIEW ),
+            JobTasks::AJAX_FINISHED_JOB_PREVIEW,
+            $job->id
+        );
+    }
+
+    /**
+     * Builds a list of filter objects.
+     *
+     * @return array
+     */
+    public static function getJobFilters()
+    {
+        $filters = [];
+        $currentFilter = (string) WordPressApi::getInput('filter');
+        $filterConfig = [
+            'All' => '',
+            'Accepted' => 'accepted',
+            'Finished' => 'finished',
+            'Screening' => 'screening',
+            'Writing' => 'writing',
+            'Draft Ready' => 'draft_ready',
+            'Revising' => 'revising',
+            'Final Ready' => 'final_ready',
+            'In Progress' => 'in_progress',
+            'Needs Review' => 'needs_review',
+        ];
+
+        array_walk($filterConfig, function ($slug, $name) use ($currentFilter, &$filters) {
+            $filters[] = new Jobs\Filter($name, $slug, $slug == $currentFilter);
+        });
+
+        return $filters;
+    }
+
+    /**
+     * Attempts to get the first post id from the given collection of post ids
+     * associated with the given job id.
+     *
+     * @param  object $job
+     * @param  array  $postIds
+     *
+     * @return string|null
+     */
+    public static function getPostIdForJob($job, array $postIds)
+    {
+        if (isset($postIds[$job->id])) {
+            return (string) array_shift($postIds[$job->id]);
+        }
+
+        return null;
+    }
+
+    /**
+     * Fetches all the jobs based on the current query, then renders a list view
+     * in the WordPress admin screen.
+     *
+     * @return string
+     */
+    public static function render()
+    {
+        wp_enqueue_style('thickbox');
+        wp_enqueue_script('thickbox');
+        $options = [];
+
+        try {
+            $paged = WordPressApi::getInput('paged');
+            $filter = WordPressApi::getInput('filter');
+
+            $jobUrl = implode('/', array_filter(['jobs', $filter]));
+            $jobUrl .= sprintf('?%s', http_build_query([
+                'next_cursor' => $paged
+            ]));
+
+            $result = Http::getResponse($jobUrl, Http::GET);
+
+            $postIds = [];
+
+            if (is_array($result->data)) {
+                $jobIds = array_map(function ($job) {
+                    return $job->id;
+                }, $result->data);
+
+                $postIds = WordPressApi::getPostIdsByProjectIds($jobIds);
+            }
+
+            $options['paginatedJobs'] = $result;
+            $options['postIds'] = $postIds;
+        } catch (AccessTokenIsUnauthorized $e) {
```

Comment:
Much like the comment above in the actual class, seems like nothing will happen here.

Reply:
@kkoppenhaver yeah, I am not sure what should happen here yet. Maybe nothing? The behavior seems to respond well showing an empty set.

Reply:
Yeah, probably ok for now. Just wanted to raise in unless you had plans for it and this piece was just overlooked.